homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,171 @@
+"""Summary data from Fyta."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Final
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FytaCoordinator, FytaEntity
+
+
+@dataclass(frozen=True)
+class FytaSensorEntityDescription(SensorEntityDescription):
+    """Describes Fyta sensor entity."""
+
+    value_fn: Callable[[str | int | float], str | int | float | datetime] = (
+        lambda value: value
+    )
+
+
+SENSORS: Final[list[FytaSensorEntityDescription]] = [
+    FytaSensorEntityDescription(
+        key="plant_name",
+        translation_key="fyta_plant_name",
+    ),
+    FytaSensorEntityDescription(
+        key="scientific_name",
+        translation_key="fyta_scientific_name",
+    ),
+    FytaSensorEntityDescription(
+        key="plant_status",
+        translation_key="fyta_plant_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:flower",
+    ),
+    FytaSensorEntityDescription(
+        key="temperature_status",
+        translation_key="fyta_temperature_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer-lines",
+    ),
+    FytaSensorEntityDescription(
+        key="light_status",
+        translation_key="fyta_light_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sun-clock-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="moisture_status",
+        translation_key="fyta_moisture_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:water-percent-alert",
+    ),
+    FytaSensorEntityDescription(
+        key="salinity_status",
+        translation_key="fyta_salinity_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        translation_key="fyta_temperature",
+        native_unit_of_measurement="°C",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="fyta_light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:weather-sunny",
```

Comment:
Please use icon translations

Reply:
Ok, if this is added in a subsequent PR or is this needed in the initial one?

Reply:
Can be the initial

Reply:
By having a deeper look into this, I don't think it makes sense to use translations on this, because the icon does not change. Other than for actual weather, I would not use a cloud or something a like if the light (or illumination) is low, as this is rather due to a shady location. I picked the sunny-icon, as this made some sense for me in the context (sensor shows, how much sunlight the plant receives), but I'm open for an alternative.

Reply:
It's called icon translations, but the better definition would be "referencing icons via a translation key". So it's a new way to get the icons out of the state machine.

Checkout integrations like adguard. Check their sensor.py and icons.json

Reply:
Thank you for the reference, will have a look at it. However, as explained for this icon no change of icon based on state is wanted, so I would rather not use icon translations but keep the static icon.

Reply:
That's also supported by that. Just set the default in the icons.json and that's it.